packages/nimble-components/src/patterns/button/styles.ts

```diff
@@ -78,26 +81,56 @@ export const styles = css`
     }
 
     .control:hover {
+        border-color: ${borderHoverColor};
         box-shadow: 0px 0px 0px ${borderWidth} ${borderHoverColor} inset;
-        outline: none;
+        transition: box-shadow ${mediumDelay};
     }
 
     .control${focusVisible} {
+        border-color: ${borderHoverColor};
         box-shadow: 0px 0px 0px ${borderWidth} ${borderHoverColor} inset;
-        outline: ${borderWidth} solid ${borderHoverColor};
-        outline-offset: -4px;
     }
 
     .control:active {
         box-shadow: none;
         outline: none;
+        transition: outline ${mediumDelay};
+        transition: box-shadow ${mediumDelay};
     }
 
     :host([disabled]) .control {
         box-shadow: none;
         outline: none;
     }
 
+    .control::before {
+        content: '';
+        position: absolute;
+        width: 100%;
+        height: 100%;
+        pointer-events: none;
+        box-sizing: border-box;
+        outline: none;
+        background-clip: content-box;
+        z-index: -1;
+    }
+
+    .control${focusVisible}::before {
+        outline: ${borderWidth} solid ${borderHoverColor};
+        outline-offset: -3px;
+        transition: outline 0.5s;
+        color: transparent;
```

Comment:
I had issues with the transition creating a black outline before the transition finished and the outline became green, which I explained in [this discussion](https://github.com/ni/nimble/pull/1300#discussion_r1238767173) with Jesse.

Reply:
From [mdn](https://developer.mozilla.org/en-US/docs/Web/CSS/color): 

> The color CSS property sets the foreground [color value](https://developer.mozilla.org/en-US/docs/Web/CSS/color_value) of an element's text and [text decorations](https://developer.mozilla.org/en-US/docs/Web/CSS/text-decoration), and sets the [currentcolor](https://developer.mozilla.org/en-US/docs/Web/CSS/color_value#currentcolor_keyword) value. currentcolor may be used as an indirect value on other properties and is the default for other color properties, such as [border-color](https://developer.mozilla.org/en-US/docs/Web/CSS/border-color).

The black outline appeared because in `.control::before`, the `color` variable wasn't set, so the outline would default to the `currentcolor` property which is the black colored outline we saw. By changing the color in `.control::before`, the currentcolor variable is no longer used and the default outline color is changed.